src/sentry/identity/oauth2.py

```diff
@@ -152,70 +154,38 @@ def get_oauth_data(self, payload):
 
         return data
 
-    def handle_refresh_error(self, req, payload):
-        error_name = "unknown_error"
-        error_description = "no description available"
-        for name_key in ["error", "Error"]:
-            if name_key in payload:
-                error_name = payload.get(name_key)
-                break
-
-        for desc_key in ["error_description", "ErrorDescription"]:
-            if desc_key in payload:
-                error_description = payload.get(desc_key)
-                break
-
-        formatted_error = f"HTTP {req.status_code} ({error_name}): {error_description}"
-
-        if req.status_code == 401:
-            logger.info(
-                "identity.oauth.refresh.identity-not-valid-error",
-                extra={
-                    "error_name": error_name,
-                    "error_status_code": req.status_code,
-                    "error_description": error_description,
-                    "provider_key": self.key,
-                },
-            )
-            raise IdentityNotValid(formatted_error)
-
-        if req.status_code == 400:
-            # this may not be common, but at the very least Google will return
-            # an invalid grant when a user is suspended
-            if error_name == "invalid_grant":
-                logger.info(
-                    "identity.oauth.refresh.identity-not-valid-error",
-                    extra={
-                        "error_name": error_name,
-                        "error_status_code": req.status_code,
-                        "error_description": error_description,
-                        "provider_key": self.key,
-                    },
-                )
-                raise IdentityNotValid(formatted_error)
-
-        if req.status_code != 200:
-            logger.info(
-                "identity.oauth.refresh.api-error",
-                extra={
-                    "error_name": error_name,
-                    "error_status_code": req.status_code,
-                    "error_description": error_description,
-                    "provider_key": self.key,
-                },
+    def get_refresh_token(
+        self, refresh_token, url: str, identity: Identity | RpcIdentity, **kwargs: Any
+    ) -> Response:
+        data = self.get_refresh_token_params(refresh_token, identity, **kwargs)
+
+        try:
+            req = safe_urlopen(
+                url=url,
+                headers=self.get_refresh_token_headers(),
+                data=data,
```

Comment:
Does this also need the `verify_ssl` option for gitlab? Not including this might break some self-hosted instances.

Reply:
yes, i need to add